### PR TITLE
[WIP] Add toml like configuration

### PR DESCRIPTION
**Pull request by csmoe**
Address #160 

Config format: (from https://github.com/rustwasm/wasm-pack/issues/160#issuecomment-396413387):
```toml
[package.metadata.wasm-pack.build.<build_mode>]
debug = true
features = ["console_error_panic_hook"]
rustc-opt-level = 0
wasm-opt = false
wasm-snip = false
wasm-bindgen = true
```
`build_mode = debug | profiling | production`

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
this is an awesome start @csmoe let me know if you need any help! i think the plan for landing this should be coordinated with the work in #163  and we'll need to get the other commands in there as well. i can start on that (or get others to start on that!) soon!

**Comment by mgattozzi** [comment]
@csmoe when you feel it's good for review just ping us :) I also just wanted to say I'm super stoked for this so thank you for working on it.

**Comment by fitzgen** [comment]
Is this actually still WIP? Or waiting on review?

**Comment by csmoe** [comment]
@fitzgen those changes were introduced into `manifest` locally indeed, but I don't know how to test for them since there is no [build](https://github.com/rustwasm/wasm-pack/issues/188) cmd, maybe I can take care of the splitting stuff, could you take a look at the no-response comment?https://github.com/rustwasm/wasm-pack/issues/188#issuecomment-400230511

**Comment by ashleygwilliams** [comment]
@csmoe if i am correct this is waiting on #216 ? if so, you can rebase onto that branch and then change the PR to be into that PR so we can be ready to merge when that lands (ideally today). thank you so much for all your hard work and especially for your patience!

additionally, i think we'll want some *very basic* docs and tests for this. lemme know if you want some direction on that.

**Comment by csmoe** [comment]
@ashleygwilliams yes, I'm waiting for `build`.
will restart tomorrow since it's midnight in my timezone.

File: src/manifest.rs [hunk 1]
```diff
@@ -129,9 +214,100 @@ impl CargoManifest {
...
+    }
+}
+
+impl WasmPackConfig {
+    fn from_table(table: Table) -> Result<WasmPackConfig, Error> {
+        let mut wasm_pack_config = WasmPackConfig::default();
+        for (key, value) in table {
+            match (key.as_str(), value.clone()) {
+                ("debug", Value::Boolean(b)) => wasm_pack_config.debug = From::from(b),
+                ("features", Value::Array(a)) => {
```
Comment (csmoe) [inline src/manifest.rs]: how does features work?
@ashleygwilliams  impl From\<Features\> for [CargoTomlFeaturesSection](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section)?

**Pull request closed without merging**